CHANGELOG.md

```diff
@@ -4,6 +4,7 @@
 
 ### New features
 
+* [#1712](https://github.com/bbatsov/rubocop/pull/1712): Set `Offense#corrected?` to `true`, `false`, or `nil` when it was, wasn't, or can't be auto-corrected, respectively. ([@vassilevsky][])
```

Comment:
When using the default formatter, everything will stay the same.

When using the JSON formatter, it will become possible to know if offences could have been autocorrected.

This PR started because I wanted to make better messages in this tool I wrote:

https://github.com/vassilevsky/stash_pull_request_commenter

If a reported offense could have been autocorrected, I wanted to show an additional message like "Run `rubocop -a path/to/file.rb` to autocorrect". But the values in the `corrected` field were unhelpful: `nil` if the `-a` option wasn't given, and `true` / `false` if it was.

Writing a custom formatter class wouldn't have solved the problem. `Offense#corrected?` was set to the same unhelpful values.

I think this change will be helpful for everybody who makes RuboCop-based tools. It makes returned values more semantically adequate.